### PR TITLE
Keep persistent connections in client

### LINKED ISSUE #65: Cache persistent connections in client
Follow up from https://github.com/saviorand/lightbug_http/pull/60#discussion_r1775777594

Currently we close each connection in the client. We need to implement a mechanism for managing persistent connections, potentially multiple at a time based on hostname. Might be worthwhile to investigate how other libraries manage them.

### PR DESCRIPTION
**Pull request by bgreni**
- Add support for persistent connections in client
-  Remove gojo dependency

Fixes #65

### CODE REVIEW

File: lightbug_http/client.mojo [hunk 1]
```diff
@@ -9,29 +9,42 @@ from lightbug_http.libc import (
...
+        self._connections = Dict[String, SysConnection]()
 
-    fn do(self, owned req: HTTPRequest) raises -> HTTPResponse:
+    fn __del__(owned self):
+        for conn in self._connections.values():
+            try:
+                conn[].close()
+            except:
+                pass
+
```
Comment (saviorand) [inline lightbug_http/client.mojo:46]: Should we log or throw an error here if the connection couldn't be closed? Might be an overkill though

File: lightbug_http/client.mojo [hunk 2]
```diff
@@ -83,31 +96,47 @@ struct Client:
...
         var new_buf = Bytes(capacity=default_buffer_size)
         var bytes_recv = conn.read(new_buf)
 
         if bytes_recv == 0:
-            conn.close()
+            self._close_conn(host_str)
+            if cached_connection:
+                return self.do(req^)
+            raise Error("No response recieved")
```
Comment (saviorand) [inline lightbug_http/client.mojo:123]:
```suggestion
            raise Error("No response received")
```

File: lightbug_http/client.mojo (hunk 1 above)
Comment (bgreni) [reply lightbug_http/client.mojo:46]: `__del__` can't raise, and I don't think libraries should generally print things but I do think we should be implementing some optional debug logging throughout the library. I'll add a comment noting that this is a good place to have one once we've implemented that.

**Comment by saviorand** [comment]
Double-checked everything and merged, nicely done 👏 👏

**Pull request merged**